lib/hocon/config_error.rb

```diff
@@ -32,13 +32,8 @@ class ConfigParseError < Hocon::ConfigError
   end
 
   class ConfigWrongTypeError < Hocon::ConfigError
-    # If you need the 4 or 5-arg constructor version, use 'construct' instead.
-    def initialize(origin, message, cause = nil)
-      super(origin, message, cause)
-    end
-
-    def self.construct(origin, path, expected, actual, cause = nil)
-      self.class.new(origin, "#{path} has type #{actual} rather than #{expected}", cause)
+    def self.with_expected_actual(origin, path, expected, actual, cause = nil)
+      ConfigWrongTypeError.new(origin, "#{path} has type #{actual} rather than #{expected}", cause)
```

Comment:
I'm fine with this change but curious as to the reason ... I thought the existing code here (which I just wrote) would be sufficient?